src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -3666,7 +3666,16 @@ public override void PostVisit(DottedNameNode node)
                     }
                     else
                     {
-                        SetDottedNameError(node, TexlStrings.ErrInvalidName, node.Right.Name.Value);
+                        FirstNameNode lhsFirstNameNode;
+                        if ((lhsFirstNameNode = node.Left.AsFirstName()) != null)
```

Comment:
nit:
our convention is to avoid assignment inside expressions.

Can you change to:
FirstNameNode lhsFirstNameNode = node.Left.AsFirstName();
if (lhsFirstNameNode != null)

Reply:
What about a cases where lhs is not a firstname node?
 a.b.missing
 Call(...).missing

Reply:
Generally, the pattern should be 
`if (node.Left is FirstNameNode lhsFirstNameNode)`..., combines the cast and null check.